rfcs/packaging/package-structure.md

```diff
@@ -0,0 +1,134 @@
+| `@fluentui/react-charting` | Has active work ongoing but is owned by a different team and doesn't really follow any of the newer patterns we're working on. (Once we establish a pattern for "high-value component" repos, we may split this out.) |
+| `@fluentui/react-experiments` | Weird legacy package which we don't have a good story for moving forward, and probably don't recommend (certainly don't promote) for use except by authoring teams of the individual components. It's a mix of components in active development or heavy use by partners, old trials of previous patterns we should definitely delete, and things of unknown status or abandoned. [More discussion here.](https://github.com/microsoft/fluentui/issues/16452) |
+<!-- prettier-ignore-end -->
+
+## Detailed Design or Proposal
+
+### Add status badges to all component package readmes
+
+The badges below are generated using https://shields.io/ and link to a placeholder wiki page which will later contain explanations of each status.
+
+We definitely want badges for stability, possibly with sub-badges for convergence status:
+
+- [![in development](https://img.shields.io/badge/stability-in%20development-orange)](https://github.com/microsoft/fluentui/wiki/Package-status)
+- [![preview](https://img.shields.io/badge/stability-preview-yellow)](https://github.com/microsoft/fluentui/wiki/Package-status)
+  - Possible sub-badge: [![preview (legacy)](<https://img.shields.io/badge/stability-preview%20(legacy)-yellow>)](https://github.com/microsoft/fluentui/wiki/Package-status)
+- [![stable](https://img.shields.io/badge/stability-stable-green)](https://github.com/microsoft/fluentui/wiki/Package-status)
+  - Possible sub-badge: [![stable (legacy)](<https://img.shields.io/badge/stability-stable%20(legacy)-green>)](https://github.com/microsoft/fluentui/wiki/Package-status)
+- [![deprecated](https://img.shields.io/badge/stability-deprecated-red)](https://github.com/microsoft/fluentui/wiki/Package-status)
+
+We might also want separate badges for category/convergence/??? (naming suggestions welcome)
+
+- [![legacy](https://img.shields.io/badge/convergence-legacy-lightgrey)](https://github.com/microsoft/fluentui/wiki/Package-status)
+- [![mixed](https://img.shields.io/badge/convergence-mixed-yellow)](https://github.com/microsoft/fluentui/wiki/Package-status)
+- [![done](https://img.shields.io/badge/convergence-done-green)](https://github.com/microsoft/fluentui/wiki/Package-status)
+
+Notes:
+
+- Charting and maybe experiments would probably fall into the "preview (legacy)" category
+- Suite packages should list which components are converged or not
+
+### Add `@fluentui/react-preview` and related communication channels
+
+Sometime in the next couple months after version 8 release, add a `@fluentui/react-preview` package which exports _only_ converged components which have reached "80% ready"/preview status. Notes:
+
+- Exports will go away once they're moved to the suite
```

Comment:
Once something is exported from `@fluentui/react`, it will no longer be exported from `@fluentui/react-preview`. Open to suggestions on this part though (we won't be setting up this package immediately).

Reply:
how would we communicate version of `@fluentui/react` with this workflow then ?

eg:
1. `@fluentui/react` present as is (v8). -> reexports ui fabric implementation of `react-cmp` package
2.  new converged component `@fluentui/react-cmp-new` is implemented
3.  we re-export  `@fluentui/react-cmp-new` from `@fluentui/react-preview`
4. `@fluentui/react-cmp-new` is ready to be propagated to "stable"
5. remove re-export from `@fluentui/react-preview`
6. replace re-export of `@fluentui/react-cmp` with `@fluentui/react-cmp-new` in `@fluentui/react` (@fluentui/react-cmp-new contains Breaking changes)

What will be the version of `@fluentui/react` (as the assumption is that there are gonna be breaking changes in probably every converged package) ?

Reply:
That's sort of what I was thinking, but a few differences:

1. `@fluentui/react` in version 8:. -> **generally does not re-export `@fluentui/react-cmp` packages** 
    - Exception: it will continue to export `@fluentui/react-focus` since we don't expect major changes there (@khmakoto correct me if I'm wrong)
    - ASAP we'll be moving the old implementations of Checkbox/Link/Slider/Pivot/Tabs out of `@fluentui/react-cmp` back into `@fluentui/react`
2.  new converged component `@fluentui/react-cmp` is implemented **(no `-new` prefix or other special naming probably)**
3.  we re-export  `@fluentui/react-cmp` from `@fluentui/react-preview`
4. `@fluentui/react-cmp` is ready to be propagated to "stable"
5. remove re-export from `@fluentui/react-preview`
6. ~~replace re-export of `@fluentui/react-cmp` with `@fluentui/react-cmp-new`~~ **Export `@fluentui/react-cmp`** in `@fluentui/react` **and if needed, move old `cmp` implementation elsewhere (possibly `@fluentui/react/lib/compat`). `@fluentui/react` major bumps due to breaking changes.**

For versioning, any time a component package which is exported from `@fluentui/react` major bumps, `@fluentui/react` must also major bump. This is an additional motivation for not exporting converged components from the suite until we're sure they're ready. (I wasn't planning on getting into versioning at this time since it's been extensively discussed elsewhere--I can fill you in on this and/or add it to a document later.)

Reply:
> move old cmp implementation elsewhere (possibly @fluentui/react/lib/compat

This feels like going back where we were/are. I had impression we don't wanna introduce common antipatterns like nested paths (private APIs) - which is also introducing another maintenance costs.

> For versioning, any time a component package which is exported from @fluentui/react major bumps, @fluentui/react must also major bump. This is an additional motivation for not exporting converged components from the suite until we're sure they're ready.

Got it, In that case I'd update the proposal that we will re-export via `@fui/react` only when all components are converged thus having 1 major bump for `@fui/react`. Otherwise it is confusing IMO.

Reply:
This is all good stuff.

I don't think we need the interim step of -preview, though. Currently, releasing unfinished ideas has caused us a lot of difficulty since we have deps out in the wild. Instead, I'd propose if someone wants to test something in development, they grab a nightly build / beta and try it. This way, we don't have to add proper maintenance overhead during iteration. We also don't need to move things around in between dev and stable.

>proposal that we will re-export via @fui/react only when all components are converged thus having 1 major bump for @fui/react

I like this, this is like the v9 idea I proposed above. Keep all iterations in our repo until we're ready, then release 9. If people need to test parts as we develop/iterate, then grab a nightly beta of the package you want to test. No strings attached and we won't support it long term until it hits major.

The one difference I'll say compared to Martin's proposal here is that I think once we have an initial set of components and stable library architecture, we can likely make many feature releases which introduce new components.  I'm not sure that waiting for _all_ to be done is a requirement prior to making a major release.

Reply:
Couple suggested tweaks or questions

####  `@fluentui/react/lib/compat` exports
Rather than use the `@fluentui/react/lib/compat` exports, can folks just update to `@fluentui/react-cmp` ahead of time, then when we major release switch the imports from `@fluentui/react-cmp` to `@fluentui/react`? 

I ask because as part of reducing the cost of partner major version upgrades, I would really like to get to the point where teams can do incremental pre-work prior to taking a major that drastically reduces the amount of work needed to do when they actually perform the major upgrade. This was one of the original purposes of the `@fluentui/react/lib/compat` export except in the opposite direction (upgrade first, fix later). 

#### "-components" suite package instead of "-preview" 
- Given we're advocating for a "-components" suite package. Would we also re-export `@fluentui/react-cmp` from `@fluentui/react-components`?

Reply:
@levithomason 
> The one difference I'll say compared to Martin's proposal here is that I think once we have an initial set of components and stable library architecture, we can likely make many feature releases which introduce new components. I'm not sure that waiting for all to be done is a requirement prior to making a major release.

Agree (and that's what I meant as well but didn't wrote ha) !